src/com/projectkorra/projectkorra/waterbending/ice/PhaseChange.java

```diff
@@ -407,7 +407,7 @@ public void melt(final Block b) {
 					new TempBlock(b, Material.AIR.createBlockData(), 120 * 1000L);
 				} else {
 					snow.setLayers(snow.getLayers() - 1);
```

Comment:
Just remove this line so it doesn't double reduce the layers. Don't reuse the same blockdata